src/codeu/chat/client/commandline/Chat.java

```diff
@@ -47,11 +49,21 @@ public Chat(Context context) {
   // is willing to take another command, the function will return true. If
   // the system wants to exit, the function will return false.
   //
-  public boolean handleCommand(String line) {
+  public boolean handleCommand(String line) throws IOException {
```

Comment:
No.

    git checkout persistent_storage # if necessary
    git fetch
    git merge origin master

Do this first. This is what I mean by "merge commits from the latest master into this branch." After that I'll review your actual changes.

The reason I'm asking you to do this is because it's harder to separate your changes from changes already in master. (Look at "Files changed.")